### DISCUSSION TITLE
Support for 3rd party JWT verification

### DISCUSSION THREAD

**Discussion by GrimLothar** [supabase/supabase#8883] [General]
Hi! I'm not sure if this is supported yet or not. But I couldn't make it work myself so I thought I'd ask.

I'm trying to use a 3rd party auth provider to have my users log in with either an email, or their Solana wallets if they have one.
The provider is called Web3auth, and they have a guide here showing how to integrate with Firebase: https://web3auth.io/docs/guides/firebase

The part where I got stuck is with the `JWK Endpoint` and `JWT Validation` fields.
<img width="801" alt="Screen Shot 2022-09-09 at 3 42 45 PM" src="https://user-images.githubusercontent.com/90475821/189421681-04139cb8-7238-4516-8097-18a04979d17c.png">
In some other discussion I found this endpoint that might have something to do with it: `https://<project_id>.supabase.co/.well-known/jwks.json` but the popup says that is an `Invalid JWK endpoint`

Any help regarding this is appreciated.

I also opened a discussion on Web3Auth's side to see if they can help as well: https://github.com/orgs/Web3Auth/discussions/582

Thank you!

**Comment by anoushk1234** [comment]
We are facing the same issue, please look into this

**Comment by marcus-sa** [comment]
Any updates..?
